### PR TITLE
Add middleware alternative

### PR DESCRIPTION
**Pull request by Zuluru**
This completes the resolution of issue #28.

### CODE REVIEW

File: docs/Middleware/Ajax.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+## Usage
+Load the Ajax middleware:
+```php
+$middlewareQueue->->add(AjaxMiddleware::class);
+```
+
+As with the component, you can pass the settings either directly inline here, or use Configure to set them globally.
+
+```php
+$middlewareQueue->->add(new AjaxMiddleware(['viewClass' => 'App.Ajax']))
```
Comment (dereuromark) [inline docs/Middleware/Ajax.md]: Do we need App. prefix? I thought just Ajax works.

**Comment by dereuromark** [comment]
Nice work. CS fail, otherwise we can merge soon.

File: docs/Middleware/Ajax.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+This middleware provides an alternative to the component, and is intended for applications which have migrated to the
+middleware stack provided in CakePHP 3.5 and above.
+Features and configuration are intended to be the same between the two methods, you should only need one or
+the other.
+Refer to [Component documentation](../Component/Ajax.md) for those details.
+
+## Usage
+Load the Ajax middleware:
+```php
+$middlewareQueue->->add(AjaxMiddleware::class);
```
[the remark this answers is not in the document]
Comment (Zuluru) [reply docs/Middleware/Ajax.md]: Changed to use the entire namespace in the `add` call.

File: docs/Middleware/Ajax.md (hunk 1 above)
Comment (Zuluru) [reply docs/Middleware/Ajax.md]: Yep. Changed to "MyAjax" to make it clear that it's a user-provided class.

File: docs/Middleware/Ajax.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dereuromark) [reply docs/Middleware/Ajax.md]: Still contains ->->

[the remark this answers is not in the document]
Comment (Zuluru) [reply docs/Middleware/Ajax.md]: Hah. Read your initial comment in email, and thought it was some weird formatting glitch from Github....

File: src/Middleware/AjaxMiddleware.php [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+		$respondAsAjax = $this->_config['autoDetect'] && $this->_isActionEnabled($request) && $request->is('ajax');
+		if ($respondAsAjax) {
+			EventManager::instance()->on('Controller.beforeRender', [$this, 'beforeRender']);
+		}
+
+		$response = $next($request, $response);
+
+		if ($respondAsAjax) {
+			EventManager::instance()->off('Controller.beforeRender', [$this, 'beforeRender']);
+			if ($this->_config['resolveRedirect'] && $response->hasHeader('Location')) {
```
Comment (dereuromark) [inline src/Middleware/AjaxMiddleware.php]: Are you sure `$response->hasHeader()` method exists?
From the looks into it, only request has this method?

**Pull request merged**